cpp/src/c_api/graph_mg.cpp

```diff
@@ -96,49 +135,27 @@ struct create_graph_functor : public cugraph::c_api::abstract_functor {
         edge_type_id_t>>
         new_edge_types{std::nullopt};
 
-      rmm::device_uvector<vertex_t> edgelist_srcs(src_->size_, handle_.get_stream());
-      rmm::device_uvector<vertex_t> edgelist_dsts(dst_->size_, handle_.get_stream());
```

Comment:
Maybe something outside the scope of this PR, but are we planning to shuffle edges in the C-API in the future? If we perform shuffle in the C-API, should we still need to take multiple arrays from the python layer?

Reply:
We do shuffle the edges in the C API currently.

Perhaps @VibhuJawa can offer some insight into the need for multiple arrays, but I believe this is an artifact of dask and how the arrays are distributed on the workers.  It is possible for there to be multiple device arrays on the same worker.  This new mechanism allows the concatenation of these multiple arrays to occur in the C API rather than in dask/cudf.

Reply:
We can do it in `dask-cudf` too once we remove the  `num_edges`  signature (which this PR does). 

https://github.com/rapidsai/cugraph/blob/8c104a52bbece783f91001c64cc6bf0732f10594/python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py#L134

If we don't have to get  `num_edges` , we can just do this in `cuDF/cuPY` one array at a time and the resulting overhead will be  `2x(array_memory)` instead of `2x(dataframe_memory)` .